force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +144,36 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        // Query to check if the account exists
+        List<Account> accountList = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        Account newAccount;
+    
+        // If the account does not exist, create a new one
+        if (accountList.isEmpty()) {
+            newAccount = new Account(Name = accountName);
+            upsert newAccount; // Upsert to insert the account if it doesn't exist
```

Comment:
I see why you used `upsert` here (because the method instructions hinted at it but it wasn't super clear). But because we absolutely know we're inserting a new Account, we should use `insert` instead:
```suggestion
            insert newAccount; // Upsert to insert the account if it doesn't exist
```